server/utils/queries/authorFilters.js

```diff
@@ -2,35 +2,48 @@ const Sequelize = require('sequelize')
 const Database = require('../../Database')
 
 module.exports = {
+  /**
+   * Get authors total count
+   * @param {string} libraryId
+   * @returns {{id:string, name:string, count:number}}
+   */
+  async getAuthorsTotalCount(libraryId) {
+    const authorsCount = await Database.authorModel.count({
+      where: {
+        libraryId: libraryId
+      }
+    });
+    return authorsCount;
+  },
+
   /**
    * Get authors with count of num books
    * @param {string} libraryId 
    * @returns {{id:string, name:string, count:number}}
    */
   async getAuthorsWithCount(libraryId) {
-    const authors = await Database.authorModel.findAll({
-      where: [
-        {
-          libraryId
-        },
-        Sequelize.where(Sequelize.literal('count'), {
-          [Sequelize.Op.gt]: 0
-        })
-      ],
+    const authors = await Database.bookAuthorModel.findAll({
+      include: [{
+        model: Database.authorModel,
+        as: 'author', // Use the correct alias as defined in your associations
+        attributes: ['name'],
+        where: {
+          libraryId: libraryId
+        }
+      }],
       attributes: [
-        'id',
-        'name',
-        [Sequelize.literal('(SELECT count(*) FROM bookAuthors ba WHERE ba.authorId = author.id)'), 'count']
+        'authorId',
+        [Sequelize.fn('COUNT', Sequelize.col('authorId')), 'count']
       ],
-      order: [
-        ['count', 'DESC']
-      ]
+      group: ['authorId', 'author.id'], // Include 'author.id' to satisfy GROUP BY with JOIN
+      order: [[Sequelize.literal('count'), 'DESC']],
+      limit: 10
```

Comment:
Might be a good idea to make the `limit` an argument to the function in case the limit should be configurable somewhere? Could still default to 10 because that's what is on the stats page.

Reply:
Sure. Just fixed that. I've added the "limit" parameter and also set the call from librarycontroller.js to be 10.